source/numerics.tex

```diff
@@ -6791,60 +6790,61 @@
 
 \indexlibrary{\idxcode{valarray}!constructor}%
 \begin{itemdecl}
-explicit valarray(size_t);
+explicit valarray(size_t n);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-The array created by this constructor has a length equal to the value of the argument.
+\effects
+Constructs a \tcode{valarray} that has length \tcode{n}.
 The elements of the array are value-initialized~(\ref{dcl.init}).
 \end{itemdescr}
 
 \indexlibrary{\idxcode{valarray}!constructor}%
 \begin{itemdecl}
-valarray(const T&, size_t);
+valarray(const T& v, size_t n);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-The array created by this constructor has a length equal to the second
-argument.
-The elements of the array are initialized with the value of the first argument.
+\effects
+Constructs a \tcode{valarray} that has length \tcode{n}.
+The elements of the array are initialized with \tcode{v}.
 \end{itemdescr}
 
 \indexlibrary{\idxcode{valarray}!constructor}%
 \begin{itemdecl}
-valarray(const T*, size_t);
+valarray(const T* p, size_t n);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-The array created by this constructor has a length equal to the second
-argument
-\tcode{n}.
+\requires
+The number of values pointed to by \tcode{p} is
+equal to or greater than \tcode{n}.
```

Comment:
We could add a cross-reference to 8.3.4 [dcl.array] after this particular "array" to disambiguate. Done.